src/app/shared/models/action-trigger.ts

```diff
@@ -1,5 +1,43 @@
+import { KeyAlias } from 'src/app/keyboard/value-objects/key-alias'
+
 export interface ActionTrigger<Data = unknown> {
   id: string
   actionId: string
   data: Data
 }
+
+export interface KeyboardTriggerData {
+  keys: string[]
+  actionArguments?: unknown
+}
+
+export interface KeyboardAndSelectionTriggerData extends KeyboardTriggerData {
+  selected?: boolean
+}
+
+export interface CreateActionTrigger<Data = unknown> {
+  actionId: string
+  data: Data
+}
```

Comment:
This interface seems redundant. It's a copy of `ActionTrigger` with just one less attribute.
It is not possible for you to use the `ActionTrigger` interface instead?

I'm assuming that it's because that you don't have an `id` for the ActionTrigger because it's the backend that is creating that, but you could just set the id to an empty string. The endpoint for creating an ActionTrigger is ignoring the `id` anyway if it receives one:
![image](https://github.com/tv2/ng-sofie/assets/23049647/b02dabb5-7b7f-4190-8806-a0134df94afc)

Reply:
Done